cnd/src/http_api/routes/index.rs

```diff
@@ -170,17 +270,64 @@ impl From<Body<Halight, Herc20>> for HalightHerc20 {
     }
 }
 
-impl<A, B> Body<A, B> {
-    fn to_created_swap<CA, CB>(&self, swap_id: LocalSwapId) -> CreatedSwap<CA, CB>
-    where
-        CA: From<A>,
-        CB: From<B>,
-        Self: Clone,
-    {
+impl From<Body<Herc20, Hbit>> for Herc20Hbit {
+    fn from(body: Body<Herc20, Hbit>) -> Self {
+        Self {
+            ethereum_expiry: body.alpha.absolute_expiry.into(),
+            erc20_amount: body.alpha.amount,
+            token_contract: body.alpha.contract_address,
+            bitcoin_expiry: body.beta.absolute_expiry.into(),
+            bitcoin_amount: *body.beta.amount,
+        }
+    }
+}
+
+impl From<Body<Hbit, Herc20>> for HbitHerc20 {
+    fn from(body: Body<Hbit, Herc20>) -> Self {
+        Self {
+            bitcoin_expiry: body.alpha.absolute_expiry.into(),
+            bitcoin_amount: *body.alpha.amount,
+            ethereum_expiry: body.beta.absolute_expiry.into(),
+            erc20_amount: body.beta.amount,
+            token_contract: body.beta.contract_address,
+        }
+    }
+}
+
+trait ToCreatedSwap<A, B> {
```

Comment:
Haha, it keeps coming back XD

You don't need this trait and any of its implementations with the generic function that used to be on line 173 :)

Reply:
It's literally in every PR we open. Sad days.